sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -1110,6 +1110,14 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
     }
     assert(e.getMessage.contains("Cannot create encoder for Option of Product type"))
   }
+
+
+  test ("SPARK-17460: the sizeInBytes in Statistics shouldn't overflow") {
+    val ds = (0 to 10000).map( i =>
+      (i, Seq((i, Seq((i, "This is really not that long of a string")))))).toDS()
+    val sizeInBytes = ds.logicalPlan.statistics.sizeInBytes
```

Comment:
Without the change, we got the following error because of overflow:

```
-1890686892 was not greater than 0
ScalaTestFailureLocation: org.apache.spark.sql.DatasetSuite$$anonfun$65 at (DatasetSuite.scala:554)
org.scalatest.exceptions.TestFailedException: -1890686892 was not greater than 0
```